### ISSUE TITLE
Possible Bug in 05_network-io/5-10_tcp-server.asciidoc code.

### ISSUE THREAD

**Issue by KevinSerafini** [clojure-cookbook/clojure-cookbook#484]
If I load the code for `serve-persistent` as written, my telnet session closes immediately after it receives the response. It appears that the problem is cased by having the second `with-open` inside the `while` loop. I changed the code a bit and it appears to work as expected:

```
(defn serve-persistent [port handler]
  (let [running (atom true)]
    (future
      (with-open [server-sock (ServerSocket. port)
                  sock (.accept server-sock)]
         (while @running
            (let [msg-in (receive sock)
                  msg-out (handler msg-in)]
              (send sock msg-out)))))
      running))
```

Also note that `send` conflicts with `clojure.core/send`, so in my code, I've renamed it to `send-msg`. I didn't change it here to match what is above.

Thanks,
Kevin

**Comment by rkneufeld** [comment]
I haven't run this in a bit, but I suspect you may be right. Would you be interested in submitting a PR to update the recipe (and letting me know how to confirm)?

**Comment by KevinSerafini** [comment]
I have a CL in my local github desktop client. I assume that if I push to master it will require your approval. Is that correct?

(Sorry for the slow turnaround. Life's been busy...)

**Comment by rkneufeld** [comment]
Not quite, if you're unfamiliar with PRs you'll want to check out <https://help.github.com/articles/using-pull-requests/>
